backend/canvas_app_explorer/alt_text_helper/background_tasks/canvas_tools_alt_text_scan.py

```diff
@@ -0,0 +1,297 @@
+    logger.info(f"Starting fetch_and_scan_course for course_id: {task.get('course_id')}")
+      # mark CourseScan as running (create if missing)
+    course_id = int(task.get('course_id'))
+
+    # adding a status before start of the scan, if this DB action failed no need to stop next steps of fetching content images
+    update_course_scan(course_id, 'running')
+
+    # Fetch course content using the manager
+    user_id = task.get('user_id')
+    req_user: User = get_user_model().objects.get(pk=user_id)
+    canvas_callback_url = task.get('canvas_callback_url')
+    logger.info(f"canvas_callback_url: {canvas_callback_url}")
+    # Create a request factory and build the request since this is a background task request won't have a user session
+    factory = RequestFactory()
+    request: Request = factory.get('/oauth/oauth-callback')
+    request.user = req_user
+    request.build_absolute_uri = lambda path: canvas_callback_url
+    session = SessionStore()
+    session['course_id'] = course_id
+    session.save()
+    request.session = session
+    try:
+        canvas_api: Canvas =  MANAGER_FACTORY.create_manager(request).canvas_api
+    except (InvalidOAuthReturnError, Exception) as e:
+        logger.error(f"Error creating Canvas API for course_id {course_id}: {e}")
+        CanvasOAuth2Token.objects.filter(user=request.user).delete()
+        update_course_scan(course_id, 'failed')
+        return
+    # Fetch full course details to ensure attributes like course_code are present for logging
+    course: Course = Course(canvas_api._Canvas__requester, {'id': course_id})
+
+    start_time: float = time.perf_counter()
+    results = get_courses_images(course)
+    logger.info(f"Fetching course {course_id} content images took {time.perf_counter() - start_time:.2f} seconds")
+    unpack_and_store_content_images(results, course)
+
+    
+@async_to_sync
+async def get_courses_images(course: Course):
+    results = await asyncio.gather(
+        fetch_content_items_async(get_assignments, course),
+        fetch_content_items_async(get_pages, course),
+        return_exceptions=True,
+    )
+    logger.debug("raw results from gather: %s", results)
+    return results
+    
+def unpack_and_store_content_images(results, course: Course):
+     # unpack results (assignments, pages) and handle exceptions returned by gather. gather maintain call order
+    assignments, pages = results
+
+    error_when_fetching_images = False
+    if isinstance(assignments, Exception):
+        logger.error("Error occurred fetching assignments: %s", assignments)
+        error_when_fetching_images = True
+        assignments = []
+    if isinstance(pages, Exception):
+        logger.error("Error occurred fetching pages: %s", pages)
+        error_when_fetching_images = True
+        pages = []
+    
+    if error_when_fetching_images:
+        # DB call would be needed to update CourseScan status to 'failed' here. 
+        update_course_scan(course.id, 'failed')
```

Comment:
If images fetch fails ( partial or full failure) will set the CourseScan to table to a failed status.